eva_submission/ena_retrieval.py

```diff
@@ -0,0 +1,117 @@
+import os.path
+
+import requests
+from ebi_eva_common_pyutils.config import cfg
+from ebi_eva_common_pyutils.logger import logging_config as log_cfg
+from ebi_eva_common_pyutils.metadata_utils import get_metadata_connection_handle
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query, execute_query
+
+from eva_submission.assembly_taxonomy_insertion import download_xml_from_ena
+
+logger = log_cfg.get_logger(__name__)
+
+
+def files_from_ena(search_term):
+    xml_root = download_xml_from_ena(f'https://www.ebi.ac.uk/ena/browser/api/xml/textsearch?result=analysis&query={search_term}')
+    analyses = xml_root.xpath('/ANALYSIS_SET/ANALYSIS')
+    analysis_files = {}
+    for analysis in analyses:
+
+        files = analysis.xpath('FILES/FILE')
+        file_dicts = []
+        for file in files:
+            file_dicts.append({'filename': file.attrib['filename'], 'filetype': file.attrib['filetype'],
+                               'md5': file.attrib['checksum'], 'analysis_accession': analysis.attrib['accession']})
+        analysis_files[analysis.attrib['accession']] = file_dicts
+    return analysis_files
+
+
+def get_file_from_md5(md5):
+    with get_metadata_connection_handle(cfg['maven']['environment'], cfg['maven']['settings_file']) as conn:
+        query = f"select file_id from file where file_md5='{md5}'"
+        rows = get_all_results_for_query(conn, query)
+        file_ids = [file_id for file_id, in rows]
+        if len(file_ids) > 1:
+            raise ValueError(f'Multiple file found with md5 {md5}')
+        elif len(file_ids) == 0:
+            return None
+        return file_ids[0]
+
+
+def create_file(file_dict):
```

Comment:
I would rename this to `insert_file_into_evapro` or something like that, and maybe similarly for `create_file_in_analysis`; I don't know if others would have this problem but I got confused that this was actually creating a file in the filesystem or something.